### ISSUE TITLE
Provide 'disabled_filetype' option

### ISSUE THREAD

**Issue by htfy96** [Shougo/deoplete.nvim#110]
I switched from vim to neovim several days ago, and this nice plugin really helps me a lot. 

However, deoplete's autocmds often confict with those of other completion engine(e.g ycm), therefore some strange problems may happen. 

Since youcompleteme would not be supported, I decided to enable youcompleteme only for `cpp`, `c` and `py` files, which was quite easy by `let g:ycm_filetype_whitelist = { 'cpp' : 1, 'c' : 1, 'python' : 1}`, but I found it difficult to disable deoplete for these filetypes.

Then I modified `init.vim` and `handlers.vim` to add a `disabled_filetype` option. Now `autocmd BufEnter,FileType * filetype_check` is installed by `handlers#_init`, and `filetype_check` registers/removes autocmds based on `filetype`.

Maybe it would be better if this option were provided. Thanks. :+1:

**Comment by wsdjeg** [comment]
can you try with 
```vim
let g:deoplete#sources = {}
let g:deoplete#sources.cpp=[]
let g:deoplete#sources.c=[]
let g:deoplete#sources.python=[]
```

**Comment by htfy96** [comment]
@wsdjeg This doesn't work because it only disable parsing `cpp`,`c` and `python` files instead of clearing deoplete's `autocmd`s

**Comment by Shougo** [comment]
You can use `b:deoplete_disable_auto_complete` for it.
But my recommendation is to use clang_complete or vim-marching for C/C++ completion.

**Comment by wsdjeg** [comment]
i think deoplete is much better than ycm,even if the filetype is c or c++

**Comment by balta2ar** [comment]
> i think deoplete is much better than ycm,even if the filetype is c or c++

@wsdjeg Why? Besides of being asynchronous, what are the benefits?

**Comment by wsdjeg** [comment]
if you are using deoplete,you will find the diff

**Comment by balta2ar** [comment]
@wsdjeg I'm not using deoplete yet, but I'd like to know its pros and cons before I switch to it. So, could you summarize, please? One of the reasons to switch to it (that I'm aware of) is that completion sources are implemented in Python3. Are there any significant reasons?

**Comment by wsdjeg** [comment]
shougo's complete plugin  can provide more than one source at the same time,for example,completion form tags buffrs omni and so on

**Comment by Shougo** [comment]
> One of the reasons to switch to it (that I'm aware of) is that completion sources are implemented in Python3. Are there any significant reasons?

YCM sources are Python2(and C).
deoplete is Python3.  I think Python3 is better if you want to use Python.

In YCM, all sources are built-in.
But deoplete is not.  You can create and add the sources without including.

**Comment by balta2ar** [comment]
Thanks, @Shougo, @wsdjeg! Another important feature of YCM for me is GoToDefinition. How do I do that with deoplete?

**Comment by wsdjeg** [comment]
I need this feature too,now I am using vim-javacomplete2 + unite-tag,but it is not the best way ！

**Comment by Shougo** [comment]
> Thanks, @Shougo, @wsdjeg! Another important feature of YCM for me is GoToDefinition. How do I do that with deoplete?

It seems nice feature.  But I think it is not completion plugin work.
It should be resolved by another plugin.

**Comment by wsdjeg** [comment]
yes ,agree with @shougo
